### ISSUE TITLE
Default background flash

### ISSUE THREAD

**Issue by jddamas** [processing/processing#3134]
Hi,

I'm using a background (0) on my sketch, but when I run it, the first thing it appears on the sketch window is a flash of the default background color (whitish). Only after a small amount time it goes to the desired background color.

I've implemented frame.setBackground(new java.awt.Color(0,0,0)) first thing on my setup() and that reduces the time of the default background flash to less than 50ms.

It would be nice to assume the background color before displaying the window.

I'm using Processing 3.0 a5 on the latest mac OS. The renderer is the default one and P2D or P3D only make the flash longer.

Thanks.

**Comment by benfry** [comment]
Lots of work has gone into reducing flicker like this over the years, but much of it has been upended with more recent code that redoes the graphics handling. 

These are tedious things to debug (across platforms) and get right; and help here is appreciated.

**Comment by benfry** [comment]
Do you have a minimal sketch that shows the problem?

**Comment by federicobond** [comment]
I do have a minimal sketch that reproduces a similar problem (OS X 10.10.3, latest Processing from repo). There is some heavy flickering going on in the rectangle on top of the image.

````processing
PImage img = loadImage("PATH_TO_ANY_JPG_OR_PNG");

void setup() {
  size(500, 500);
}

void draw() {
  image(img, 0, 0, 500, 500);
  rect(200, 200, 200, 200);
}
````

**Comment by benfry** [comment]
@fredericobond actually that's a different issue but also serious. Can you open something separate for that? This particular issue is about the window flashing before showing the first frame of a sketch. What that example shows is that the screen updates and the animation thread are fighting.

**Comment by federicobond** [comment]
@benfry sure, I will open another issue for that.

**Comment by benfry** [comment]
Thanks; appreciate it.

**Comment by benfry** [comment]
(Removing the 'help' tag for now because I'm actively working on the renderer and thread code.)

**Comment by benfry** [comment]
Think I have this sorted out for the next release, which has less flicker than 3.0a5 (and much less than 2.x for that matter).
